src/tools/r2rdump/R2RHeader.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text;
+
+namespace R2RDump
+{
+    class R2RHeader
+    {
+        public const uint READYTORUN_SIGNATURE = 0x00525452; // 'RTR'
+
+        uint RelativeVirtualAddress;
+        int Offset;
+        int Size;
+
+        string SignatureString;
+        uint Signature;
+        ushort MajorVersion;
+        ushort MinorVersion;
+        uint Flags;
+        public uint NumberOfSections;
```

Comment:
It would be nice to have each of these 'fields' as 'properties', which is a special construct in C#/MSIL as compared to Java. They will behave similarly to the current 'fields' you have now, and I would just modify them to look like the following:
`public uint NumberOfSections { get; }`

Reply:
Essentially properties are a syntax-friendly way to give you a "getter and setter" that you would implement by hand in Java, and this allows you to control access of the field values better than simply accessing the variable directly.